keras_cv/models/resnet_v1.py

```diff
@@ -0,0 +1,357 @@
+    convolution layers.
+
+    For transfer learning use cases, make sure to read the [guide to transfer
+        learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+    Args:
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(1/255.0)` layer.
+        include_top: whether to include the fully-connected layer at the top of the
+            network.  If provided, num_classes must be provided.
+        num_classes: optional number of classes to classify images into, only to be
+            specified if `include_top` is True, and if no `weights` argument is
+            specified.
+        weights: one of `None` (random initialization), or a pretrained weight file
+            path.
+        input_shape: optional shape tuple, defaults to (None, None, 3).
+        pooling: optional pooling mode for feature extraction
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be the 4D tensor output
+                of the last convolutional block.
+            - `avg` means that global average pooling will be applied to the output
+                of the last convolutional block, and thus the output of the model will
+                be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        name: (Optional) name to pass to the model.  Defaults to "{name}".
+    Returns:
+      A `keras.Model` instance.
+"""
+
+
+def V1Block(filters, kernel_size=3, stride=1, conv_shortcut=True, name=None):
+    """A residual block.
+    Args:
+      x: input tensor.
+      filters: integer, filters of the bottleneck layer.
+      kernel_size: default 3, kernel size of the bottleneck layer.
+      stride: default 1, stride of the first layer.
+      conv_shortcut: default True, use convolution shortcut if True,
+          otherwise identity shortcut.
+      name: string, block label.
+    Returns:
+      Output tensor for the residual block.
+    """
+    if name is None:
+        name = f"v1_block_{backend.get_uid('v1_block')}"
+
+    def apply(x):
+        if conv_shortcut:
+            shortcut = layers.Conv2D(4 * filters,
+                                     1,
+                                     strides=stride,
+                                     name=name + "_0_conv")(x)
+            shortcut = layers.BatchNormalization(axis=BN_AXIS,
+                                                 epsilon=1.001e-5,
+                                                 name=name + "_0_bn")(shortcut)
+        else:
+            shortcut = x
+
+        x = layers.Conv2D(filters, 1, strides=stride, name=name + "_1_conv")(x)
+        x = layers.BatchNormalization(axis=BN_AXIS,
+                                      epsilon=1.001e-5,
+                                      name=name + "_1_bn")(x)
+        x = layers.Activation("relu", name=name + "_1_relu")(x)
```

Comment:
Possible to delegate this to a separate function called something like `conv_bn_relu()`?  I am suggesting it because it's been used many times. Or would it make the code more complicated?